src/Components/Admin/ObjectEditor/object.js

```diff
@@ -0,0 +1,263 @@
+            getPuzzleBody();
+        }
+    }, [sceneId, objectId, puzzleBody, handleError]);
+
+    const selectPuzzleType = (obj) => {
+        if (obj) {
+            const data = puzzleBody;
+            data.componentType = obj.value;
+            setPuzzleBody(data);
+            setPuzzleType(obj.value);
+        }
+    };
+
+    const reorder = (texts, startIndex, endIndex) => {
+        const result = texts;
+        const [removed] = result.splice(startIndex, 1);
+        result.splice(endIndex, 0, removed);
+        return result;
+    };
+
+    const reorderTransitions = (sourceIndex, destinationIndex) => {
+        if (
+            sourceIndex == null ||
+            destinationIndex == null ||
+            sourceIndex === destinationIndex
+        ) {
+            return;
+        }
+
+        const reorderedList = reorder(texts, sourceIndex, destinationIndex);
+
+        setTexts([...reorderedList]);
+    };
+
+    const addTransition = () => {
+        const newTransition = {
+            text: prompt("Enter the text for the transition: "),
+            index: texts.length,
+        };
+
+        setTexts([...texts, newTransition]);
+    };
+
+    const onMoveUpClick = (index) => {
+        reorderTransitions(index, Math.max(0, index - 1));
+    };
+
+    const onMoveDownClick = (index) => {
+        reorderTransitions(index, Math.min(texts.length - 1, index + 1));
+    };
+
+    const deleteTransition = (index) => {
+        const tempTexts = JSON.parse(JSON.stringify(texts));
+        tempTexts.splice(index, 1);
+        for (let i = 0; i < tempTexts.length; i++) {
+            tempTexts[i].index = i;
+        }
+        setTexts(tempTexts);
+    };
+
+    const handleSubmit = () => {
+        const animCopy = animationsJson;
+        const puzzleBodyCopy = puzzleBody;
+        const textsCopy = texts;
+        for (let i = 0; i < textsCopy.length; i++) {
+            delete textsCopy["index"];
+        }
+        puzzleBodyCopy.jsonData.data = textsCopy;
+        animCopy.blackboardData = puzzleBodyCopy;
+        setAnimationsJson(animCopy);
+
+        const savePuzzle = async () => {
+            await editPuzzle(
+                { sceneId, objectId, isInteractable, animationsJson },
+                handleError
+            );
+        };
+
+        savePuzzle();
+    };
+
+    return (
+        <div className={classes.container}>
+            {isInteractable ? (
+                <Select
+                    value={puzzleTypeList.filter(
+                        (option) => option.value === puzzleType
+                    )}
+                    options={puzzleTypeList}
+                    placeholder="Select puzzle type..."
+                    noResultsText="No puzzle types found"
+                    searchable={true}
+                    onChange={selectPuzzleType}
+                />
+            ) : null}
+            {isInteractable && puzzleType === "text-pane" ? (
+                <div>
```

Comment:
Might want to put this into a method and return this from there, and just call that method here. When we add the other puzzle types, this render will get super confusing if it is one giant return statement.